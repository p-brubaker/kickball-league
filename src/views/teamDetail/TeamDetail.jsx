import { useParams, Link } from 'react-router-dom'
import { useState, useEffect } from 'react'
import { getTeamById } from '../../services/teams'
import PlayerList from '../playerList/PlayerList'

export default function TeamDetail() {
    const { id } = useParams()
    const [team, setTeam] = useState(null)

    useEffect(() => {
        getTeamById(id).then(res => setTeam(res))
    }, [id])

    return team ? (
        <div className="team-detail">
            <Link to={`/teams/update/${id}`}>Update this team</Link>
            <Link to={`/players/add/${id}`}>Add a player to this team</Link>
            <p>{team.name}</p>
            <p>City: {team.city}</p>
            <p>State: {team.state}</p>
            <PlayerList teamID={team.id} />
        </div>
    ) : <p>Loading...</p>
}
