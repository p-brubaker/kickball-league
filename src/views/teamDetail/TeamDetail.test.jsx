import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom' 
import TeamDetail from './TeamDetail'
import { setupServer } from 'msw/node'
import { rest } from 'msw'
import mockPlayers from '../../mocks/players.json'

const mockTeam = {
    id: 1,
    created_at: '2021-12-12T05:34:30.551Z',
    name: 'Rose City Roasters',
    city: 'Portland',
    state: 'Oregon',
    players: []
}

const server = setupServer(
    rest.get('https://paozlutduarjgvzaiduh.supabase.co/rest/v1/players', 
        (req, res, ctx) => {
            return res(ctx.json(mockPlayers))
        }
    ),
    rest.get('https://paozlutduarjgvzaiduh.supabase.co/rest/v1/teams',
        (req, res, ctx) => {
            return res(ctx.json(mockTeam))
        }
    )
)

beforeAll(() => {
    server.listen()
})

afterAll(() => {
    server.close()
})

it('should render team details and a list of that teams players', async () => {
    const { container } = render(
        <MemoryRouter>
            <TeamDetail match={{params: {id: 1}}}/>
        </MemoryRouter>
    )

    screen.getByText(/Loading.../)

    await screen.findByText(/Players/)
    expect(container).toMatchSnapshot()
})

