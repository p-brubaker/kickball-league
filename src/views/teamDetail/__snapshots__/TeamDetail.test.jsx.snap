// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render team details and a list of that teams players 1`] = `
<div>
  <div
    class="team-detail"
  >
    <a
      href="/teams/update/undefined"
    >
      Update this team
    </a>
    <a
      href="/players/add/undefined"
    >
      Add a player to this team
    </a>
    <p>
      Rose City Roasters
    </p>
    <p>
      City: 
      Portland
    </p>
    <p>
      State: 
      Oregon
    </p>
    <div
      class="players-container"
    >
      <h1>
        Players
      </h1>
      <div
        class="players-list"
      />
    </div>
  </div>
</div>
`;
